gapis/api/gles/dependencygraph2_test.go

```diff
@@ -0,0 +1,189 @@
+// Copyright (C) 2019 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gles_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/gapid/core/assert"
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/core/memory/arena"
+	"github.com/google/gapid/core/os/device"
+	"github.com/google/gapid/core/os/device/bind"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/api/gles"
+	"github.com/google/gapid/gapis/capture"
+	"github.com/google/gapid/gapis/database"
+	"github.com/google/gapid/gapis/memory"
+	"github.com/google/gapid/gapis/resolve/dependencygraph2"
+)
+
+// Test helpers
+
+// testDependencyCmdName is used as a map key to easily express
+// expected dependencies
+type testDependencyCmdName struct {
+	sourceCmdName, targetCmdName string
```

Comment:
It seems that identifying nodes based only on the command name will only work for very small tests (with just 1 call to each distinct command). But it does work for the tests here.

Reply:
Yes I also though about this, I feel we may have to enrich the way we identify a node later on. I feel like I need to face more complex captures to know what is best to do, so I'll leave this for a future change.